Cant start newest release on Windows

TAP Driver version: 9.9.2_3
These are typed in CMD as Administrator

C:\WINDOWS\system32>"C:\Program Files\Yggdrasil Network\yggdrasil-0.3.5-windows-i386.exe" -useconffile "C:\Program Files\Yggdrasil Network\config.conf"
```
2019/09/20 23:02:06 Build name: yggdrasil
2019/09/20 23:02:06 Build version: 0.3.5
2019/09/20 23:02:06 Starting up...
2019/09/20 23:02:06 Found 2 multicast interface(s)
2019/09/20 23:02:06 Starting switch
2019/09/20 23:02:06 Starting router
2019/09/20 23:02:06 Multicast discovery is enabled
2019/09/20 23:02:06 TCP admin socket listening on 127.0.0.1:9001
2019/09/20 23:02:06 netsh command: netsh interface set interface Ethernet 2 admin=DISABLED
2019/09/20 23:02:06 Listening for TCP on: [fe80::4457:9f86:12a:502c%Ethernet]:51811
2019/09/20 23:02:06 Listening for TCP on: [fe80::996e:c618:cd29:6408%Ethernet 2]:51812
2019/09/20 23:02:08 netsh command: netsh interface set interface Ethernet 2 admin=ENABLED
2019/09/20 23:02:09 Interface name: Ethernet 2
2019/09/20 23:02:09 Interface IPv6: [REDACTED]/7
2019/09/20 23:02:09 Interface MTU: 65535
2019/09/20 23:02:09 Startup complete
2019/09/20 23:02:09 Your IPv6 address is [REDACTED]
2019/09/20 23:02:09 Your IPv6 subnet is [REDACTED]/64
```
C:\WINDOWS\system32>"C:\Program Files\Yggdrasil Network\yggdrasil-0.3.8-windows-i386.exe" -useconffile "C:\Program Files\Yggdrasil Network\config.conf"
```
2019/09/20 23:07:18 Build name: yggdrasil
2019/09/20 23:07:18 Build version: 0.3.8
2019/09/20 23:07:18 Starting up...
2019/09/20 23:07:18 Starting switch
2019/09/20 23:07:18 Starting router
2019/09/20 23:07:18 Startup complete
2019/09/20 23:07:18 Listening for TCP on: [fe80::4457:9f86:12a:502c%Ethernet]:51913
2019/09/20 23:07:18 Listening for TCP on: [fe80::996e:c618:cd29:6408%Ethernet 2]:51914
2019/09/20 23:07:18 TCP admin socket listening on 127.0.0.1:9001
2019/09/20 23:07:21 Interface name: Ethernet 2
2019/09/20 23:07:21 Interface IPv6: [REDACTED]/7
2019/09/20 23:07:21 Interface MTU: 65535
2019/09/20 23:07:21 Your IPv6 address is [REDACTED]
2019/09/20 23:07:21 Your IPv6 subnet is [REDACTED]/64
panic: runtime error: index out of range

goroutine 76 [running]:
github.com/yggdrasil-network/yggdrasil-go/src/tuntap.(*TunAdapter).readerPacketHandler(0x11478bcc, 0x1150a700)
        /home/circleci/project/src/tuntap/iface.go:113 +0x84c
created by github.com/yggdrasil-network/yggdrasil-go/src/tuntap.(*TunAdapter).reader
        /home/circleci/project/src/tuntap/iface.go:253 +0x8b
```

--- a/src/tuntap/iface.go
+++ b/src/tuntap/iface.go
@@ -148,6 +148,11 @@ func (tun *TunAdapter) _handlePacket(recvd []byte, err error) {
 	// Offset the buffer from now on so that we can ignore ethernet frames if
 	// they are present
 	bs := recvd[offset:]
+	// Check if the packet is long enough to detect if it's an ICMP packet or not
+	if len(bs) < 7 {
+		tun.log.Traceln("TUN/TAP iface read undersized unknown packet, length:", len(bs))
+		return
+	}
 	// If we detect an ICMP packet then hand it to the ICMPv6 module
 	if bs[6] == 58 {
 		// Found an ICMPv6 packet - we need to make sure to give ICMPv6 the full
@@ -175,6 +180,7 @@ func (tun *TunAdapter) _handlePacket(recvd []byte, err error) {
 	if bs[0]&0xf0 == 0x60 {
 		// Check if we have a fully-sized IPv6 header
 		if len(bs) < 40 {
+			tun.log.Traceln("TUN/TAP iface read undersized ipv6 packet, length:", len(bs))
 			return
 		}
 		// Check the packet size
@@ -188,6 +194,7 @@ func (tun *TunAdapter) _handlePacket(recvd []byte, err error) {
 	} else if bs[0]&0xf0 == 0x40 {
 		// Check if we have a fully-sized IPv4 header
 		if len(bs) < 20 {
+			tun.log.Traceln("TUN/TAP iface read undersized ipv6 packet, length:", len(bs))
 			return
 		}
 		// Check the packet size

--- a/src/tuntap/iface.go
+++ b/src/tuntap/iface.go
@@ -194,7 +194,7 @@ func (tun *TunAdapter) _handlePacket(recvd []byte, err error) {
 	} else if bs[0]&0xf0 == 0x40 {
 		// Check if we have a fully-sized IPv4 header
 		if len(bs) < 20 {
-			tun.log.Traceln("TUN/TAP iface read undersized ipv6 packet, length:", len(bs))
+			tun.log.Traceln("TUN/TAP iface read undersized ipv4 packet, length:", len(bs))
 			return
 		}
 		// Check the packet size

--- a/src/util/cancellation.go
+++ b/src/util/cancellation.go
@@ -11,8 +11,8 @@ import (
 // This is and is similar to a context, but with an error to specify the reason for the cancellation.
 type Cancellation interface {
 	Finished() <-chan struct{} // Finished returns a channel which will be closed when Cancellation.Cancel is first called.
-	Cancel(error) error // Cancel closes the channel returned by Finished and sets the error returned by error, or else returns the existing error if the Cancellation has already run.
-	Error() error // Error returns the error provided to Cancel, or nil if no error has been provided.
+	Cancel(error) error        // Cancel closes the channel returned by Finished and sets the error returned by error, or else returns the existing error if the Cancellation has already run.
+	Error() error              // Error returns the error provided to Cancel, or nil if no error has been provided.
 }
 
 // CancellationFinalized is an error returned if a cancellation object was garbage collected and the finalizer was run.

